src/components/PopupBase/__tests__/PopupBase.spec.tsx

```diff
@@ -14,28 +16,32 @@ describe('<PopupBase> that adds basic anchor/popup functionality to rendered com
     const popupRendererMock = ({ setPopupVisibility }) => (
         <PopoverDummy setPopupVisibility={setPopupVisibility} />
     );
-    let wrapper;
+    let view: RenderResult;
 
     describe('rendering', () => {
         beforeEach(() => {
-            wrapper = mount(
+            view = render(
                 <PopupBase anchorRenderer={anchorRendererMock} popupRenderer={popupRendererMock} />
             );
         });
 
         it('should render with minimal props correctly', () => {
-            expect(toJson(wrapper)).toMatchSnapshot();
-            expect(wrapper.find('Popover')).toHaveLength(0);
+            expect(view.container).toMatchSnapshot();
+            expect(screen.queryByRole('group')).not.toBeInTheDocument();
         });
-        it('should render popup when requested', () => {
+
+        it('should render popup when requested', async () => {
+            const user = userEvent.setup();
             // trigger setPopupVisibility(true) through our dummy component
-            wrapper.find('button').simulate('click');
+            await user.click(screen.getByRole('button'));
 
-            expect(toJson(wrapper)).toMatchSnapshot();
-            expect(wrapper.find('Popover')).toHaveLength(1);
+            expect(view.container).toMatchSnapshot();
+            expect(screen.getByRole('group')).toBeInTheDocument();
         });
-        it('should render popup in portal when requested', () => {
-            wrapper = mount(
+
+        it('should render popup in portal when requested', async () => {
+            const user = userEvent.setup();
+            view = render(
```

Comment:
Do we need this render? Looks like it is the same component as the one that is being render first in the beforeEach. But for some reason if we need we should use rerender.

Reply:
:+1:

Reply:
:+1: